src/adera/Machines/Rocket.cpp

```diff
@@ -139,35 +111,28 @@ void SysMachineRocket::update_physics()
         {
             Percent *percent = std::get_if<Percent>(throttle);
 
-            float thrust = 10.0f; // temporary
+            float thrustMag = 10.0f; // temporary
 
             //std::cout << percent->m_value << "\n";
+            Matrix4 relTransform{};
+            SysPhysics_t::find_rigidbody_ancestor(rScene, ent, &relTransform);
 
-            Vector3 thrustVec = compTf->m_transform.backward()
-                                    * (percent->m_value * thrust);
-
-            SysPhysics_t::body_apply_force(*compRb, thrustVec);
-        }
-
-
-
-        // this is suppose to be gimbal, but for now it applies torque
+            Vector3 thrustDir = relTransform.transformVector(Vector3{0.0f, 0.0f, 1.0f});
+            Vector3 thrust = thrustMag*percent->m_value * thrustDir;
+            Vector3 location = relTransform.translation();
```

Comment:
I realized that I don't use the center of mass offset here anywhere, and I don't think `find_rigidbody_ancestor()` gets it. Everything behaves physically find but I'll have to test it to make sure. I think it needs to be added.